src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs

```diff
@@ -111,6 +109,24 @@ public static void CreateDirectory(string fullPath)
             // descriptor and set it in he call to CreateDirectory.
             Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = default;
 
+            if (dirSecurity != null)
+            {
+                secAttrs.nLength = (uint)Marshal.SizeOf(secAttrs);
+
+                // For ACLs, get the security descriptor from the FileSecurity
+                byte[] sd = dirSecurity.GetSecurityDescriptorBinaryForm();
+
+                unsafe
+                {
+                    fixed (byte* sdPtr = sd)
+                    {
+                        byte* bytesOnStack = stackalloc byte[sd.Length];
+                        Buffer.MemoryCopy(sdPtr, bytesOnStack, sd.Length, sd.Length);
```

Comment:
Framework had MemCpy, Core has MemoryCopy, with a different parameter order. Was this the correct method to use?

Reply:
Neither. Pin it.

Reply:
The Framework code isn't right. Copying to the stack is unnecessary- it could have just been pinned there as well.